src/main/java/seedu/taassist/model/session/SessionData.java

```diff
@@ -0,0 +1,54 @@
+package seedu.taassist.model.session;
+
+import static seedu.taassist.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Objects;
+
+import seedu.taassist.model.student.StudentSessionData;
+import seedu.taassist.model.uniquelist.Identity;
+
+/**
+ * Represents a session data of student in a module.
+ */
+public class SessionData implements Identity<SessionData> {
+
+    private final Session session;
+    private final StudentSessionData data;
+
+    /**
+     * Constructs a {@code SessionData} with the given session and student session data.
+     */
+    public SessionData(Session session, StudentSessionData data) {
+        requireAllNonNull(session, data);
+        this.session = session;
+        this.data = data;
+    }
+
+    public Session getSession() {
+        return session;
+    }
+
+    public StudentSessionData getData() {
+        return data;
+    }
+
+    @Override
+    public boolean isSame(SessionData other) {
+        return other == this
+                || (other != null && other.session.equals(this.session));
```

Comment:
```suggestion
                || (other != null && other.session.isSame(this.session));
```

Perhaps use `isSame`? This will avoid issues down the line if we extend `Session` with additional fields.